### PR TITLE
build(ci): Use codecov GH action instead of bash script

### PR DESCRIPTION
**Pull request by billyvg**
This uses a GH Action instead of bash script. The bash script has been deprecated. See https://github.com/codecov/codecov-action

* Change our uploads by adding a flag for frontend or backend. 
* Only upload coverage of acceptance tests on backend changes (unsure how this will affect codecov for frontend only changes)

This does *NOT* fix the spiky reporting of coverage (which I believe is due to flakey tests, eg, 3 of 4 test instances fail, but we still report 3 of them to codecov.

Note, due to a [bug in codecov](https://github.com/codecov/uploader/issues/475), we have to unset `NODE_OPTIONS`

### CODE REVIEW

File: .github/actions/artifacts/action.yml [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
     - name: Upload to codecov
-      shell: bash
-      run: |
-        coverage_files=$(ls .artifacts/*coverage.xml || true)
-        if [[ -n "$coverage_files" || -f .artifacts/coverage/cobertura-coverage.xml ]]; then
-          bash <(curl -s https://codecov.io/bash) -v
-        fi
+      uses: codecov/codecov-action@v2
+      env:
+          NODE_OPTIONS: ""
```
Comment (armenzg) [inline .github/actions/artifacts/action.yml:20]: Why empty?

Comment (billyvg) [reply .github/actions/artifacts/action.yml]: Oops, added a comment

File: Makefile [hunk 2]
```diff
@@ -161,7 +161,7 @@ test-plugins:
 
 test-relay-integration:
 	@echo "--> Running Relay integration tests"
-	pytest tests/relay_integration -vv
+	pytest tests/relay_integration -vv --cov . --cov-report="xml:.artifacts/relay.coverage.xml" --junit-xml=".artifacts/relay.junit.xml"
```
Comment (armenzg) [inline Makefile:164]: Can I assume this is intentional?

**Review by armenzg — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (billyvg) [reply Makefile:164]: Yeah I initially refactored these paths and noticed relay wasn't outputting coverage stats.

**Pull request merged**